awx/ui/src/contexts/Session.js

```diff
@@ -89,11 +89,15 @@ function SessionProvider({ children }) {
       return () => {};
     }
 
-    const calcRemaining = () =>
-      parseInt(sessionTimeout, 10) - new Date().getTime();
+    const calcRemaining = () => {
+      Math.max(0, parseInt(sessionTimeout, 10) - new Date().getTime());
```

Comment:
let's use luxon for current time

Reply:
@kialam, I can update to use luxon, but this is how we dealt with the same conversion on `Base.js`.

```
defaultHttp.interceptors.response.use((response) => {
  const timeout = response?.headers['session-timeout'];
  if (timeout) {
    const timeoutDate = new Date().getTime() + timeout * 1000;
    updateStorage(SESSION_TIMEOUT_KEY, String(timeoutDate));
  }
  return response;
});
```

Reply:
Yup, we are using a date time library so I think it makes sense to start using it moving forward. The luxon equivalent is `Datetime.now().milliseconds`.